coalib/settings/ConfigurationGathering.py

```diff
@@ -403,7 +403,7 @@ def get_config_directory(section):
     if 'project_dir' in section:
         return path(section.get('project_dir'))
 
-    config = os.path.abspath(section.get('files', '').origin)
+    config = os.path.abspath(str(section.get('files', '').origin))
```

Comment:
when `__str__` is changed to also give the line and col, this is going to break.

See comment about @property for `SourcePosition.origin`